libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendTaskModuleUrlResponse.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * @module botbuilder-dialogs-adaptive-teams
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    BoolExpression,
+    BoolExpressionConverter,
+    Expression,
+    StringExpression,
+    StringExpressionConverter,
+} from 'adaptive-expressions';
+import { TaskModuleContinueResponse, TaskModuleResponse, TaskModuleTaskInfo } from 'botbuilder';
+import { Converter, ConverterFactory, DialogConfiguration, DialogContext, DialogTurnResult } from 'botbuilder-dialogs';
+import { BaseSendTaskModuleContinueResponse } from './baseSendTaskModuleContinueResponse';
+import { BaseTeamsCacheInfoResponseDialog } from './baseTeamsCacheInfoResponseDialog';
+
+export interface SendTaskModuleUrlResponseConfiguration extends DialogConfiguration {
+    disabled?: boolean | string | BoolExpression;
+    url?: string | Expression | StringExpression;
+    fallbackUrl?: string | Expression | StringExpression;
+}
+
+/**
+ * Send a simple message task module response.
+ */
+export class SendTaskModuleUrlResponse
+    extends BaseSendTaskModuleContinueResponse
+    implements SendTaskModuleUrlResponseConfiguration {
+    /**
+     * Class identifier.
+     */
+    public static $kind = 'Teams.SendTaskModuleUrlResponse';
+
+    /**
+     * Gets or sets an optional expression for the Url of the Task Module response.
+     */
+    public url: StringExpression;
+
+    /**
+     * Gets or sets an optional expression for the Fallback Url the Task Module Task Info response.
+     */
+    public fallbackUrl: StringExpression;
+
+    public getConverter(property: keyof SendTaskModuleUrlResponseConfiguration): Converter | ConverterFactory {
+        switch (property) {
+            case 'disabled':
+                return new BoolExpressionConverter();
+            case 'url':
+            case 'fallbackUrl':
+                return new StringExpressionConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Called when the dialog is started and pushed onto the dialog stack.
+     *
+     * @param {DialogContext} dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param {object} options Optional, initial information to pass to the dialog.
+     * @returns {Promise<DialogTurnResult>} A promise representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: Record<string, unknown>): Promise<DialogTurnResult> {
+        if (this.disabled && this.disabled.getValue(dc.state)) {
```

Comment:
Since you've got optional chaining down, use it here! `if (this.disabled?.getValue(dc.state)) { return ... }`